### PR TITLE
Take clearButton into account in editingRect

### LINKED ISSUE #147: Clear button is gets overlapped with text?
When clearButtonMode is set to .whileEditing, The text gets overlapped with text.
In my opinion, the culprit is this overridden method. It doesn't consider the clearButtonMode while returning the CGRect. The width should be less while clearButtonMode is .whileEditing.

```swift
override open func editingRect(forBounds bounds: CGRect) -> CGRect {

        let rect = CGRect(

            x: 0,

            y: titleHeight(),

            width: bounds.size.width,

            height: bounds.size.height - titleHeight() - selectedLineHeight

        )

        return rect

    }
```

### PR DESCRIPTION
**Pull request by k0nserv**
Resolves #147

### CODE REVIEW

File: Sources/SkyFloatingLabelTextField.swift [hunk 1]
```diff
@@ -478,12 +478,14 @@ open class SkyFloatingLabelTextField: UITextField { // swiftlint:disable:this ty
...
+
         let rect = CGRect(
-            x: 0,
-            y: titleHeight(),
-            width: bounds.size.width,
-            height: bounds.size.height - titleHeight() - selectedLineHeight
+            x: superRect.origin.x,
+            y: titleHeight,
+            width: superRect.size.width,
+            height: bounds.size.height - titleHeight - selectedLineHeight
```
Comment (k0nserv) [inline Sources/SkyFloatingLabelTextField.swift]: Suppose this should be `superRect.size.height` instead of `bounds.size.height`

**Comment by k0nserv** [comment]
Yeah I suppose we should take all values from superRect?

**Comment by bogren** [comment]
Yes I think that would be better :)

**Comment by k0nserv** [comment]
Done

**Comment by k0nserv** [comment]
@bogren Does this look good now?

**Pull request merged**